File: post-day-4/src/topic_pkg/src/topic_publisher.cpp
#include "ros/ros.h"
#include "topic_pkg/Topic.h"

int main(int argc, char** argv) {
    ros::init(argc, argv, "topic_publisher");
    
    ros::NodeHandle nh;
    ros::Publisher topic_publisher = nh.advertise<topic_pkg::Topic>("topic_pkg", 1000);

    ros::Rate loop_rate(1);

    while (ros::ok()) {
        topic_pkg::Topic topic_to_send;

        // Data untuk di publish
        topic_to_send.usia = 19;
        topic_to_send.tinggi = 160.5;
        topic_to_send.nama = "Generic Name";
        topic_to_send.kehadiran = false;

        topic_publisher.publish(topic_to_send);

        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: post-day-4/src/topic_pkg/script/topic_publisher.py
#!/usr/bin/env python3

import rospy
from topic_pkg.msg import Topic

def talker():
    publisher = rospy.Publisher("topic_pkg_py", Topic, queue_size=10)
    rospy.init_node('topic_pkg_py', anonymous=True)

    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        topic_to_send = Topic()

        topic_to_send.usia = 20
        topic_to_send.tinggi = 180.41
        topic_to_send.nama = "John Doe"
        topic_to_send.kehadiran = True

        publisher.publish(topic_to_send)

        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass